src/test/net_tests.cpp

```diff
@@ -725,8 +731,79 @@ BOOST_AUTO_TEST_CASE(ipv4_peer_with_ipv6_addrMe_test)
 
     // suppress no-checks-run warning; if this test fails, it's by triggering a sanitizer
     BOOST_CHECK(1);
+
+    // Cleanup, so that we don't confuse other tests.
+    {
+        LOCK(cs_mapLocalHost);
+        mapLocalHost.erase(addr_in_mapLocalHost);
+    }
 }
 
+BOOST_AUTO_TEST_CASE(get_local_addr_for_peer_port)
+{
+    // Test that GetLocalAddrForPeer() properly selects the address for self-advertise:
```

Comment:
39b4a0c309c5f63bdf0bdbde0d76ee3edce3ac19 "for self-advertisement" or "to self-advertise"

Reply:
Done.